rsconnect/api.py

```diff
@@ -713,6 +887,179 @@ def filter_out_server_info(**kwargs):
+
+    def _get_canonical_request_signature(self, request):
+        result = hmac.new(self._key, request.encode(), hashlib.sha256).hexdigest()
+        return base64.b64encode(result.encode()).decode()
+
+    def get_extra_headers(self, url, method, body):
+        canonical_request_method = method.upper()
+        canonical_request_path = parse.urlparse(url).path
+        canonical_request_date = datetime.datetime.utcnow().strftime("%a, %d %b %Y %H:%M:%S GMT")
+
+        # get request checksum
+        md5 = hashlib.md5()
+        body = body or b""
+        body_bytes = body if isinstance(body, bytes) else body.encode()
+        md5.update(body_bytes)
+        canonical_request_checksum = md5.hexdigest()
+
+        canonical_request = self._get_canonical_request(
+            canonical_request_method, canonical_request_path, canonical_request_date, canonical_request_checksum
+        )
+
+        signature = self._get_canonical_request_signature(canonical_request)
+
+        return {
+            "X-Auth-Token": "{0}".format(self._token),
+            "X-Auth-Signature": "{0}; version=1".format(signature),
+            "Date": canonical_request_date,
+            "X-Content-Checksum": canonical_request_checksum,
+        }
+
+    def get_application(self, application_id):
+        return self.get("/v1/applications/{}".format(application_id))
+
+    def create_application(self, account_id, application_name):
+        application_data = {
+            "account": account_id,
+            "name": application_name,
+            "template": "shiny",
+        }
+        return self.post("/v1/applications/", body=application_data)
+
+    def get_accounts(self):
+        return self.get("/v1/accounts/")
+
+    def _get_applications_like_name_page(self, name: str, offset: int):
+        return self.get(
+            "/v1/applications?filter=name:like:{}&offset={}&count=100&use_advanced_filters=true".format(name, offset)
+        )
+
+    def create_bundle(self, application_id: int, content_type: str, content_length: int, checksum: str):
+        bundle_data = {
+            "application": application_id,
+            "content_type": content_type,
+            "content_length": content_length,
+            "checksum": checksum,
+        }
+        return self.post("/v1/bundles", body=bundle_data)
+
+    def set_bundle_status(self, bundle_id, bundle_status):
+        return self.post("/v1/bundles/{}/status".format(bundle_id), body={"status": bundle_status})
+
+    def deploy_application(self, bundle_id, app_id):
+        return self.post("/v1/applications/{}/deploy".format(app_id), body={"bundle": bundle_id, "rebuild": False})
+
+    def get_task(self, task_id):
+        return self.get("/v1/tasks/{}".format(task_id), query_params={"legacy": "true"})
+
+    def get_current_user(self):
+        return self.get("/v1/users/me")
+
+    def wait_until_task_is_successful(self, task_id, timeout=60):
```

Comment:
Builds can sometimes take quite a bit so I would probably up this...

```suggestion
    def wait_until_task_is_successful(self, task_id, timeout=180):
```